src/network/tc/qdisc.c

```diff
@@ -178,6 +178,33 @@ int qdisc_configure(Link *link, QDiscs *qdisc) {
         return 0;
 }
 
+int qdisc_section_verify(QDiscs *qdisc, bool *has_network_emulator, bool *has_token_buffer_filter) {
+        assert(qdisc);
+        assert(has_network_emulator);
+        assert(has_token_buffer_filter);
+        assert(!*has_network_emulator || !*has_token_buffer_filter);
+
+        if (section_is_invalid(qdisc->section))
+                return -EINVAL;
+
+        if (qdisc->has_network_emulator && qdisc->has_token_buffer_filter)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "%s: TrafficControlQueueingDiscipline section has both NetworkEmulator and TokenBufferFilter settings. "
+                                         "Ignoring [TrafficControlQueueingDiscipline] section from line %u.",
+                                         qdisc->section->filename, qdisc->section->line);
+
+        if ((*has_network_emulator && qdisc->has_token_buffer_filter) ||
+            (*has_token_buffer_filter && qdisc->has_network_emulator))
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "%s: Two kind of TrafficControlQueueingDiscipline sections are defined. "
+                                         "Ignoring [TrafficControlQueueingDiscipline] section from line %u.",
+                                         qdisc->section->filename, qdisc->section->line);
+
+        *has_network_emulator = qdisc->has_network_emulator;
+        *has_token_buffer_filter = qdisc->has_token_buffer_filter;
+        return 0;
```

Comment:
```
[TrafficControlQueueingDiscipline]
Parent=root
TokenBufferFilterRate=1M
TokenBufferFilterBurst=32K
TokenBufferFilterLatencySec=400 ms
```

```
[TrafficControlQueueingDiscipline]
Parent=clsact
NetworkEmulatorDelaySec=50ms
NetworkEmulatorDelayJitterSec=10ms
NetworkEmulatorLossRate=20%
NetworkEmulatorPacketLimit=100
```

This is valid 
```
qdisc tbf 8005: dev dummy99 root refcnt 2 rate 1Mbit burst 32000b lat 400.0ms 
qdisc netem ffff: dev dummy99 parent ffff:fff1 limit 100 delay 50.0ms  10.0ms loss 20%
```


But  this below is not.
```
[TrafficControlQueueingDiscipline]
Parent=root
TokenBufferFilterRate=1M
NetworkEmulatorLossRate=20%
NetworkEmulatorPacketLimit=100
``` 

We can apply  TBF and NetEM if  one section `Parent=root` and another `Parent=clsact` but if different we can't.

Reply:
I see. Thanks. I will update so.